service/Abstractions/Pipeline/DataPipeline.cs

```diff
@@ -249,6 +249,13 @@ public string GetHandlerOutputFileName(IPipelineStepHandler handler, int index =
     [JsonPropertyName("completed_steps")]
     public List<string> CompletedSteps { get; set; } = new();
 
+    /// <summary>
+    /// Tells if the pipeline failed.
+    /// </summary>
+    [JsonPropertyOrder(6)]
+    [JsonPropertyName("failed")]
+    public bool Failed { get; set; } = false;
```

Comment:
how do you ensure this is consistent? 

for instance, if ContentStorage becomes unavailable and the pipeline job expires, the value will remain `false` and never change to `true` because there's nothing left running to retry.